rclcpp/include/rclcpp/node_interfaces/node_base.hpp

```diff
@@ -18,13 +18,26 @@
 #include <memory>
 #include <string>
 #include <vector>
+#include <unordered_map>
+#include <mutex>
+#include <iostream>
 
 #include "rcl/node.h"
 #include "rclcpp/context.hpp"
 #include "rclcpp/macros.hpp"
 #include "rclcpp/node_interfaces/node_base_interface.hpp"
 #include "rclcpp/visibility_control.hpp"
 
+#if 1
+#define DPRINT(x) std::cout << "DPRINT: " << x << std::endl;
+#else
+#define DPRINT(x)
+#endif
+
+typedef std::function<void (rclcpp::CallbackGroup::SharedPtr)> CallbackGroupFunction;
+
+class map_of_mutexes;
```

Comment:
These should both be in the namespace.  I would suggest putting them into the `rclcpp::node_interfaces` namespace, but other namespaces under `rclcpp` would work as well.

Also, make sure to `#include <functional>` at the top of this file to get `std::function`.